src/coreComponents/fieldSpecification/FieldSpecificationManager.hpp

```diff
@@ -233,80 +222,97 @@ class FieldSpecificationManager : public dataRepository::Group
+    constexpr std::array< char const *, 2 > groupsToSkip =
+    {
+      ElementRegionManager::groupKeyStruct::elementRegionsGroup(),
+      ElementRegionBase::viewKeyStruct::elementSubRegions()
+    };
+
+    for( std::size_t pathLevel = 0; pathLevel < targetPath.size(); ++pathLevel )
+    {
+      // Skip certain groups in the tree by jumping directly to them
+      for( char const * const groupName : groupsToSkip )
+      {
+        if( dataRepository::Group * const subGroup = targetGroup->getGroupPointer( groupName ) )
+        {
+          targetGroup = subGroup;
+        }
+      }
+
+      // Also skip them in the object path being processed
+      if( std::find( groupsToSkip.begin(), groupsToSkip.end(), targetPath[pathLevel] ) != groupsToSkip.end() )
+      {
+        continue;
+      }
+
+      // If the next token is a list, process it recursively and stop
+      if( targetPath[pathLevel][0] == '{' )
+      {
+        array1d< string > subGroupNames;
+        LvArray::input::stringToArray( subGroupNames, targetPath[pathLevel] );
+        for( string const & subGroupName : subGroupNames )
+        {
+          applyToPathRecursive( &targetGroup->getGroup( subGroupName ),
+                                targetPath.subspan( pathLevel + 1 ),
+                                fs, std::forward< LAMBDA >( lambda ) );
+        }
+        return;
+      }
+
+      targetGroup = &targetGroup->getGroup( targetPath[pathLevel] );
+    }
+    applyOnTargetRecursive( *targetGroup, fs, std::forward< LAMBDA >( lambda ) );
+  }
+
+  template< typename BCTYPE, typename LAMBDA >
+  void applyOnTargetRecursive( dataRepository::Group & target,
                                BCTYPE const & fs,
-                               string const & targetName,
-                               LAMBDA && lambda
-                               ) const
+                               LAMBDA && lambda ) const
   {
-    if( ( target.getParent().getName() == ElementRegionBase::viewKeyStruct::elementSubRegions()
-          || target.getName() == "nodeManager"
-          || target.getName() == "FaceManager"
-          || target.getName() == "edgeManager" ) // TODO these 3 strings are harcoded because for the moment, there are
-                                                 // inconsistencies with the name of the Managers...
-        && target.getName() != ObjectManagerBase::groupKeyStruct::setsString()
-        && target.getName() != ObjectManagerBase::groupKeyStruct::neighborDataString() )
+    // Set-based application makes sense only for classes derived from ObjectManagerBase...
+    // except for ElementRegionManager and ElementRegionBase, which are intermediate layers.
+    ObjectManagerBase const * const manager = dynamicCast< ObjectManagerBase * >( &target );
+    if( manager != nullptr
+        && target.getName() != MeshLevel::groupStructKeys::elemManagerString
+        && target.getParent().getName() != ElementRegionManager::groupKeyStruct::elementRegionsGroup() )
```

Comment:
Maybe
```suggestion
    if( auto manager = dynamicCast< ObjectManagerBase const * >( &target )
        && target.getName() != MeshLevel::groupStructKeys::elemManagerString
        && target.getParent().getName() != ElementRegionManager::groupKeyStruct::elementRegionsGroup() )
```
So `null` ptr `manager` cannot be accessed in other scope in future devs.

Reply:
I've tried it and it's a compiler error, because `&&` takes priority over assignment, and then it tries to implicitly convert pointer to `bool` or something. Basically, we can't do it until C++17 where init-statement can be separated by `;` from the branch condition.

Reply:
Even with the parenthesis that solve the world? 🤣

Reply:
> Even with the parenthesis that solve the world?

Sadly, that would not be [valid syntax](https://en.cppreference.com/w/cpp/language/if) :(
```
condition - one of
 * expression which is contextually convertible to bool
 * declaration of a single non-array variable with a brace-or-equals initializer
```